rs/nervous_system/tools/release-runscript/src/utils.rs

```diff
@@ -92,3 +92,134 @@ pub(crate) fn press_enter_to_continue() -> Result<()> {
+    }
+
+    println!("{}", "GitHub CLI is configured ✓".bright_green());
+
+    Ok(())
+}
+
+pub(crate) fn commit_all_into_branch(branch: &str) -> Result<()> {
+    let ic = ic_dir();
+
+    {
+        // Check if branch exists
+        let output = Command::new("git")
+            .current_dir(&ic)
+            .args(["branch", "--list", branch])
+            .output()?;
+
+        let branch_exists = !String::from_utf8_lossy(&output.stdout).trim().is_empty();
+        if branch_exists {
+            if input_yes_or_no(
+                &format!("Branch '{}' already exists. Delete it?", branch),
+                false,
+            )? {
+                // Delete the branch
+                let output = Command::new("git")
+                    .current_dir(&ic)
+                    .args(["branch", "-D", branch])
+                    .output()?;
+                if !output.status.success() {
+                    bail!(
+                        "Failed to delete branch: {}",
+                        String::from_utf8_lossy(&output.stderr)
+                    );
+                }
+                println!(
+                    "{}",
+                    format!("Deleted existing branch '{}'", branch).bright_blue()
+                );
+            } else {
+                bail!("Cannot continue with existing branch");
+            }
+        }
+    }
+
+    let output = Command::new("git")
+        .current_dir(&ic)
+        .args(["checkout", "-b", branch])
+        .output()?;
+    if !output.status.success() {
+        return Err(
+            anyhow::anyhow!("{}", String::from_utf8_lossy(&output.stderr))
+                .context("Failed to create branch"),
+        );
+    }
+
+    let output = Command::new("git")
+        .current_dir(&ic)
+        .args(["add", "."])
+        .output()?;
+    if !output.status.success() {
+        return Err(
+            anyhow::anyhow!("{}", String::from_utf8_lossy(&output.stderr))
+                .context("Failed to add all files to branch"),
+        );
+    }
+
+    let output = Command::new("git")
+        .current_dir(&ic)
+        .args(["commit", "-m", "chore(nervous-system): update changelog"])
+        .output()?;
+    if !output.status.success() {
+        return Err(
+            anyhow::anyhow!("{}", String::from_utf8_lossy(&output.stderr))
+                .context("Failed to commit all files to branch"),
+        );
+    }
+
+    Ok(())
+}
+
+pub(crate) fn create_pr(title: &str, body: &str) -> Result<url::Url> {
+    // push the current branch to the remote repository
+    // e.g. git push --set-upstream origin <branch-name>
+    let branch = Command::new("git")
+        .arg("branch")
+        .arg("--show-current")
+        .output()?;
+    let branch = String::from_utf8(branch.stdout)?;
+    let output = Command::new("git")
+        .arg("push")
+        .arg("--set-upstream")
+        .arg("origin")
+        .arg("--force")
```

Comment:
NCR

I guess this is to create a branch in GH, right? I did not know this.

Another way to do that:

```
.arg(&format!("{}:{}", branch, branch))
```

This is maybe more explicit, because to me, it is not clear what is being "forced".

In my personal workflow, my GH branch also has my name in it, which would be done more or less like so:

```
.arg(&format!("{}:{}-{}", branch, branch, username))
```

However, this might mess up `gh pr create ...`.

Reply:
In theory we could easily ask it what branch name you'd like to use, or have it automatically use your name from git